extension/observer/cfgardenobserver/extension.go

```diff
@@ -4,33 +4,337 @@
 package cfgardenobserver // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/observer/cfgardenobserver"
 
 import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"sync"
 	"time"
 
+	"code.cloudfoundry.org/garden"
+	gardenClient "code.cloudfoundry.org/garden/client"
+	gardenConnection "code.cloudfoundry.org/garden/client/connection"
+	"github.com/cloudfoundry/go-cfclient/v3/client"
+	"github.com/cloudfoundry/go-cfclient/v3/config"
+	"github.com/cloudfoundry/go-cfclient/v3/resource"
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/extension"
+	"go.uber.org/zap"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/observer"
 )
 
 type cfGardenObserver struct {
 	*observer.EndpointsWatcher
+	cancel context.CancelFunc
+	config *Config
+	ctx    context.Context
```

Comment:
[Contexts shouldn't be stored in `structs`.](https://go.dev/blog/context-and-structs) Is there some other way to handle the necessary functionality?

Reply:
Good point, I honestly just did the same as the dockerobserver. Will try to find an alternative.